packages/entando-apimanager/modules/api/apiManager.js

```diff
@@ -142,6 +118,94 @@ const normalizeErrorMessage = (message) => {
   return 'serverError';
 };
 
+export const authIntercept = () => {
+  if ('currentUser' in store.getState() && getRefreshToken()) {
+    authorize(getRefreshToken(), getAuthenticationToken());
+  }
+};
+
+const beginInterceptSetup = () => {
+  const configIntercept = {
+    // specify if request should be intercepted here we intercept
+    // intercepts only if refreshTokenConfig object props is specified
+    shouldIntercept: () => refreshTokenConfig !== null,
+    // no need to change auth params, so just return as it is
+    authorizeRequest: (request, accessToken) => {
+      const authstring = request.headers.get('Authorization');
+      if (!authstring) return request;
+
+      const authtype = authstring.split(' ')[0];
+      if (authtype !== 'Basic') {
+        request.headers.set('Authorization', `Bearer ${accessToken}`);
+      }
+      return request;
+    },
+    // create a `Request` object for fetch on renewing access token
+    createAccessTokenRequest: (token) => {
+      const request = refreshTokenConfig.generateParams(token);
+      return new Request(
+        getCompleteRequestUrl(request),
+        getRequestParams(request),
+      );
+    },
+    // extract the new access token from response,
+    // parseResults must have access_token and refresh_token keyprop
+    parseAccessToken: response => (
+      refreshTokenConfig.parseResults(response)
+        .then((json) => {
+          store.dispatch(setTokenCredentials(json.access_token, json.refresh_token));
+          authIntercept();
+          return json.access_token;
+        })
+    ),
+  };
+  configure(configIntercept);
+};
+
+export const config = (
+  reduxStore,
+  newLoginPage = () => {},
+  newLandingPage = () => {},
+  newRefreshTokenConfig = null,
+) => {
+  store = reduxStore;
+  landingPage = newLandingPage;
+  loginPage = newLoginPage;
+  if (
+    newRefreshTokenConfig &&
+    ('generateParams' in newRefreshTokenConfig &&
+    'parseResults' in newRefreshTokenConfig)
+  ) {
+    refreshTokenConfig = newRefreshTokenConfig;
+  }
+  beginInterceptSetup();
+  authIntercept();
```

Comment:
why are executing the intercept at the end of the setup?

Reply:
the 2 function calls were originally inside the IF block above it but I noticed when I don't provide refreshTokenConfig, fetch-token-intercept still attempts to call config functions even you don't configure and authorize it, thus throwing bunch of errors. I looked inside it's code and it shows that whether or not configuring it, it will still look for the config functions. I think a good solution for this is to rename and restruct this part to avoid confusion to others who will look into this.